client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -541,6 +542,16 @@ public static ContentType createContentType(final XContentType xContentType) {
         return ContentType.create(xContentType.mediaTypeWithoutParameters(), (Charset) null);
     }
 
+    static Request indicesExist(IndicesExistsRequest request) {
+        String endpoint = endpoint(request.indices(), Strings.EMPTY_ARRAY, "");
+        Params params = Params.builder();
+        params.withLocal(request.local());
+        params.withIndicesOptions(request.indicesOptions());
+        params.withFlatSettings(false);
+        params.withIncludeDefaults(false);
```

Comment:
sorry, I didn't realize that IndicesExistsRequest did not expose these flags, which are though supported in our rest-spec. I think that it may be better to accept GetIndexRequest here as it better represents this type of request. IndicesExistsRequest is really only used by the transport client and will be removed one day. At that point you will also have to add a withHuman method and you will be able to read the actual values for these flags from the request.

Reply:
Cool, I can do it. However, I cannot see those flags (flat_settings and include_defaults) in GetIndexRequest or its indices options (but I may be looking at the wrong place). If needed, I can of course add, along with parsing in the REST layer (if that can be considered within scope).

Reply:
oh right, `flat_settings` and `include_defaults` are REST only flags that don't apply to the transport client, only the REST layer does something with them. That is why the current requests don't support them. But now that we are reusing these same requests to make REST requests, we should add such flags. Would you mind adding those to `GetIndexRequest`?

Reply:
Aye, aye, captain!

Reply:
Added. Let me know if that is what you were looking for. I see that ClusterInfoRequest, which the GetIndexRequest inherits, has only a few fields, which appear to be more generic, so I just added it to GetIndexRequest. If this looks fine, I'll proceed with rest (`human` flag, tests, docs).

Reply:
your changes look good, you can proceed with the rest ;) thanks!

Reply:
Thank you for your patience with all the questions!